fix(types): [other] cannot find declare module types from installed packages

yuhengshen opened the pull request:
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.


`@microsoft/api-extractor` breaks TypeScript's recognition of declare module 'vue'.
​**fix**:​​ Temporarily resolve this by manually importing it. 
 
Test it with pnpm, it's working now

A better fix may exist.

yuhengshen committed:
fix(types): cannot find declare module types from installed packages

btea reviewed (approved):
Thanks!

yuhengshen commented:
@warmthsea I tested it with an empty project using this version and it worked. Have you reloaded Volar?

![image](https://github.com/user-attachments/assets/baf3dbe6-4ff8-4181-b8ce-4536deb12d4c)

yuhengshen committed:
fix(types): play global type error

yuhengshen commented:
> > @warmthsea I tested it with an empty project using this version and it worked. Have you reloaded Volar?
> 
> Yes, new project only import ` "element-plus/global"`, but it did affect the [element-plus/play](https://github.com/element-plus/element-plus/tree/dev/play)

fixed

btea commented:
It looks like `pnpm typecheck` doesn't pass.

yuhengshen committed:
fix(types): typecheck error

btea commented on packages/utils/vue3.3.polyfill.ts:
```diff
@@ -0,0 +1,51 @@
+
+type PublicOptionalKeys<T> = Exclude<keyof T, PublicRequiredKeys<T>>
+
+declare module 'vue' {
```
Perhaps a safer approach would be to remove the type declaration here and import the type directly using the file path, without having to modify `formattedText`.

yuhengshen replied:
But this approach is not ideal for future upgrades. Currently, upgrading only requires replacing instances of __ExtractPublicPropTypes and changing some files. Using type imports would necessitate manually modifying numerous files.

btea replied:
I don't think the difference is big. If upgrade in the future, you can just re-export the type here.
`export type { ExtractPublicPropTypes as __ExtractPublicPropTypes } from 'vue'`

btea replied:
WDYT？ cc  @warmthsea @Dsaquel

yuhengshen replied:
This will result in two import sources for the same type for a long time.

btea replied:
Yes, this is not a good idea either.

yuhengshen commented:
> @yuhengshen It seems to be ok for now.
Yes, I tested it with the following methods

- Using the global type.
- importing { xxx } from 'element-plus'.
- importing { xxx } from 'element-plus/es/xxx'.

btea commented:
Ok, let's merge it.

yuhengshen commented:
@warmthsea 
Sorry for reproducing this error again.

When I added the type field in tsconfig.web.json, Volar started working, but `typecheck` reported type errors.

After I fixed the type error, Volar stopped working again.

I suspect this might be a Volar bug. It seems to load the declare module definitions multiple times, which leads to a TypeScript error: Duplicate identifier '__ExtractPublicPropTypes'.

warmthsea commented:
I will submit a PR later and have a solution.
